### PR TITLE
handle market state conditions paused | open | closed

### PR DESCRIPTION
**Pull request by brossetti1**
for new markets we need to present a paused state when the market is not active.

## Description
This implementation is currently a working paused state setup only. There is a demo branch for this pr that forces all markets to paused to inspect the UI changes:

https://github.com/Kwenta/kwenta/pull/679 - demo only, all markets forced to suspended

This branch needs iteration to implement the final designs, but can serve as a basic setup in case we need to merge something before the weekend.

## Related issue
https://github.com/Kwenta/kwenta/issues/594

## Motivation and Context
I wanted to split this out into a simple implementation as a good point to iterate on, while also providing basic functionality of paused markets. There are small UI issues still and there is a decent amount of refactoring which im working on via another branch. If that branch gets into a finished state before we need to merge this, I will include those changes here.

## How Has This Been Tested?
- forced to pasued - tested deploy on https://github.com/Kwenta/kwenta/pull/679
- local dev environment

## Screenshots (if appropriate):

all screenshots from forced pause branch

Screen Shot 2022-04-13 at 7 51 23 AM
Screen Shot 2022-04-13 at 7 51 34 AM
Screen Shot 2022-04-13 at 7 51 56 AM

### CODE REVIEW

File: pages/market/[[...market]].tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ const Market: FC = () => {
 				<title>{t('futures.market.page-title', { pair: router.query.market })}</title>
 			</Head>
 			<AppLayout>
+				{/* <StyledPageContent className="paused"> */}
```
Comment (Tburm) [inline pages/market/[[...market]].tsx:25]: Can we remove this?

File: pages/market/[[...market]].tsx [hunk 2]
```diff
@@ -43,6 +44,10 @@ export default Market;
 
 const StyledPageContent = styled(PageContent)`
 	max-width: 1440px;
+
+	/* &.paused * {
+		color: ${(props) => props.theme.colors.common.secondaryGray};
+	} */
```
Comment (Tburm) [inline pages/market/[[...market]].tsx:50]: Can we remove this?

File: sections/dashboard/FuturesMarketsTable/FuturesMarketsTable.tsx [hunk 3]
```diff
@@ -269,6 +284,9 @@ const StyledTable = styled(Table)`
 const TableHeader = styled.div``;
 
 const StyledText = styled.div`
+	/* display: flex; */
+	/* align-items: center;
+	justify-content: center; */
```
Comment (Tburm) [inline sections/dashboard/FuturesMarketsTable/FuturesMarketsTable.tsx:289]: delete

File: sections/dashboard/FuturesPositionsTable/FuturesPositionsTable.tsx [hunk 4]
```diff
@@ -268,8 +281,12 @@ const StyledTable = styled(Table)`
 const TableHeader = styled.div``;
 
 const StyledText = styled.div`
+	/* display: flex;
+  align-items: center;
+	justify-content: center; */
```
Comment (Tburm) [inline sections/dashboard/FuturesPositionsTable/FuturesPositionsTable.tsx:286]: delete

File: sections/futures/MarketDetails/MarketDetails.tsx [hunk 5]
```diff
@@ -178,15 +179,24 @@ const MarketDetails: React.FC<MarketDetailsProps> = ({ baseCurrencyKey }) => {
...
+					<div key={key}>
+						<p className="heading">{key}</p>
+						<span className={spanClass}>{value}</span>
+					</div>
+				);
+			})}
 		</MarketDetailsContainer>
 	);
 };
+// color ? `value ${color} paused` : 'value paused'
```
Comment (Tburm) [inline sections/futures/MarketDetails/MarketDetails.tsx:199]:
```suggestion
```

File: queries/futures/useGetFuturesMarkets.ts [hunk 6]
```diff
@@ -38,14 +40,18 @@ const useGetFuturesMarkets = (options?: UseQueryOptions<FuturesMarket[]>) => {
...
+					const marketKey = getMarketKey(asset, network.id);
+					return utils.formatBytes32String(marketKey);
 				})
 			);
 
-			return markets
-				.filter((_: any, i: number) => suspensions[i] === false)
-				.map(
-					({
+			// const suspensionsWithReason = zip(suspensions, reasons);
```
Comment (Tburm) [inline queries/futures/useGetFuturesMarkets.ts:48]:
```suggestion
```

File: queries/futures/useGetFuturesMarkets.ts [hunk 7]
```diff
@@ -38,14 +40,18 @@ const useGetFuturesMarkets = (options?: UseQueryOptions<FuturesMarket[]>) => {
...
+
+			// console.log(suspensionsWithReason);
```
Comment (Tburm) [inline queries/futures/useGetFuturesMarkets.ts:50]:
```suggestion
```

File: queries/futures/useGetFuturesMarkets.ts [hunk 8]
```diff
@@ -8,6 +8,8 @@ import { isL2State, isWalletConnectedState, networkState } from 'store/wallet';
 import Connector from 'containers/Connector';
 import QUERY_KEYS from 'constants/queryKeys';
 import { FuturesMarket } from './types';
+import { getMarketKey } from 'utils/futures';
+// import { zip } from 'lodash';
```
Comment (Tburm) [inline queries/futures/useGetFuturesMarkets.ts:12]:
```suggestion
```

File: sections/futures/MarketOverlay/MarketOverlay.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+	const { t } = useTranslation();
+	return (
+		<OverlayContainer>
+			<Overlay>
+				<OverlayContent>
+					<StyledSvg src={PausedIcon} />
+					<StyledText>This market has been closed</StyledText>
+					{/* <StyledSubText>
+						Trading activity will resume in: <StyledTimer>12:09:22</StyledTimer>
+					</StyledSubText> */}
```
Comment (Tburm) [inline sections/futures/MarketOverlay/MarketOverlay.tsx:18]:
```suggestion
```

**Review by Tburm — APPROVED** [review]
Looks good, can we wait for some markets to be paused and test this again before merging?

**Comment by brossetti1** [comment]
closing in favor of https://github.com/Kwenta/kwenta/pull/692

**Pull request closed without merging**